src/main/java/slimeknights/tconstruct/blocks/DecorativeBlocks.java

```diff
@@ -1,137 +1,74 @@
-    registry.registerSlabsAndStairs(new Block(BlockProperties.DRIED_CLAY_BRICKS), "dried_clay_bricks");
+  public static void init() {
+    IEventBus modEventBus = FMLJavaModLoadingContext.get().getModEventBus();
+    BLOCKS.register(modEventBus);
   }
 
-  @SubscribeEvent
-  static void registerBlockItems(final RegistryEvent.Register<Item> event) {
-    BlockItemRegistryAdapter registry = new BlockItemRegistryAdapter(event.getRegistry(), TinkerRegistry.tabGeneral);
-
-    registry.registerBlockItem(clear_glass);
-    registry.registerBlockItem(white_clear_stained_glass);
-    registry.registerBlockItem(orange_clear_stained_glass);
-    registry.registerBlockItem(magenta_clear_stained_glass);
-    registry.registerBlockItem(light_blue_clear_stained_glass);
-    registry.registerBlockItem(yellow_clear_stained_glass);
-    registry.registerBlockItem(lime_clear_stained_glass);
-    registry.registerBlockItem(pink_clear_stained_glass);
-    registry.registerBlockItem(gray_clear_stained_glass);
-    registry.registerBlockItem(light_gray_clear_stained_glass);
-    registry.registerBlockItem(cyan_clear_stained_glass);
-    registry.registerBlockItem(purple_clear_stained_glass);
-    registry.registerBlockItem(blue_clear_stained_glass);
-    registry.registerBlockItem(brown_clear_stained_glass);
-    registry.registerBlockItem(green_clear_stained_glass);
-    registry.registerBlockItem(red_clear_stained_glass);
-    registry.registerBlockItem(black_clear_stained_glass);
-
-    registry.registerBlockItem(mud_bricks);
-    registry.registerBlockItem(mud_bricks_slab);
-    registry.registerBlockItem(mud_bricks_stairs);
-
-    registry.registerBlockItem(dried_clay);
-    registry.registerBlockItem(dried_clay_slab);
-    registry.registerBlockItem(dried_clay_stairs);
-
-    registry.registerBlockItem(dried_clay_bricks);
-    registry.registerBlockItem(dried_clay_bricks_slab);
-    registry.registerBlockItem(dried_clay_bricks_stairs);
-  }
+  /* Decorative Blocks */
+  public static final BlockItemObject<ClearGlassBlock> clear_glass = BLOCKS.register("clear_glass", () -> new ClearGlassBlock(BlockProperties.GENERIC_GLASS_BLOCK), generalProps);
+  public static final EnumObject<ClearStainedGlassBlock.GlassColor,ClearStainedGlassBlock> clear_stained_glass = BLOCKS.registerEnum(ClearStainedGlassBlock.GlassColor.values(), "clear_stained_glass", (color) -> new ClearStainedGlassBlock(BlockProperties.GENERIC_GLASS_BLOCK, color), generalProps);
 
-  private static void registerGlass(BaseRegistryAdapter<Block> registry) {
-    registry.register(new ClearGlassBlock(BlockProperties.GENERIC_GLASS_BLOCK), "clear_glass");
+  public static final BuildingBlockObject mud_bricks = BLOCKS.registerBuilding("mud_bricks", BlockProperties.MUD_BRICKS, generalProps);
+  public static final BuildingBlockObject dried_clay = BLOCKS.registerBuilding("dried_clay", BlockProperties.DRIED_CLAY, generalProps);
```

Comment:
Not a fan of dried clay getting walls since its supposed to be like vanilla terracotta. Maybe we should remove stairs and slabs from it honestly, but that might be something for a later commit.

Reply:
Yeah, I had an idea to have extended BuildingBlockObject for things which have a wall
and maybe fences for wood-like blocks.

Reply:
For now, might be best to ditch the walls and add them back later, I need to clean up a lot of the blocks in terms of which get walls and which get stairs and slabs.